islandora_web_archive.drush.inc

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/**
+ * @file
+ * Implementation of Drush hooks.
+ */
+
+/**
+ * Implements hook_drush_command().
+ */
+function islandora_web_archive_drush_command() {
+  return array(
+    'islandora_web_archive_migrate_png_to_screenshot' => array(
+      'bootstrap' => DRUSH_BOOTSTRAP_MAX,
+      'drupal_dependencies' => array(
+        'islandora',
+      ),
+      'description' => 'Iterates each web archive object and copies the PNG datastream into a SCREENSHOT datastream.',
+      'examples' => array(
+        'drush -u 1 islandora_web_archive_migrate_png_to_screenshot',
+      ),
+    ),
+  );
+}
+
+/**
+ * Implements hook_drush_command().
+ */
+function drush_islandora_web_archive_migrate_png_to_screenshot() {
+  // Get all islandora web archive objects.
+  $query = <<<EOQ
+    PREFIX fedora-model: <info:fedora/fedora-system:def/model#>
+    SELECT DISTINCT ?pid
+    FROM <#ri>
+    WHERE {
+      ?pid <fedora-model:hasModel> <info:fedora/islandora:sp_web_archive>
+    }
+EOQ;
+  $connection = islandora_get_tuque_connection();
+  $archive_pids = $connection->repository->ri->sparqlQuery($query);
+  module_load_include('inc', 'islandora_web_archive', 'includes/utilities');
+  foreach ($archive_pids as $pid) {
```

Comment:
Not sure if we care but foreaching here can be really expensive in memory land. Usually with these update style queries we set a limit and chunk in slices of numbers or use createdDate and order by oldest to newest to update, an example can be seen here: https://github.com/Islandora/islandora_xacml_editor/blob/7.x/api/islandora_xacml_api.drush.inc#L57.